docs/pages/inboxes/send-messages.md

```diff
@@ -36,6 +36,183 @@ try await group.send(content: "Hello everyone")
+
+### Key UX considerations
+
+- After initially sending a message optimistically, show the user an indicator that the message is still being processed. After successfully sending the message, show the user a success indicator.
+
+- If an optimistically sent message fails to send, give the user an option to retry sending the message or cancel sending. Use a try/catch block to intercept errors and allow the user to retry or cancel.
+
+### 1. Optimistically send a message locally
+
+There are two steps to optimistically send a message:
+
+1. Send the message locally so it can display immediately in the sender's UI.
+2. Send the message to the XMTP network so it can be delivered to the recipient.
+
+:::code-group
+
+```tsx [Browser]
+// Send a message optimistically (displays immediately in the sender's UI)
+conversation.sendOptimistic("Hello world");
+
+// For custom content types, specify the content type
+const customContent = { foo: "bar" };
+const contentType = { authorityId: "example", typeId: "test", versionMajor: 1, versionMinor: 0 };
+conversation.sendOptimistic(customContent, contentType);
+```
+
+```tsx [Node]
+// Send a message optimistically (displays immediately in the sender's UI)
+conversation.sendOptimistic("Hello world");
+
+// For custom content types, specify the content type
+const customContent = { foo: "bar" };
+const contentType = { authorityId: "example", typeId: "test", versionMajor: 1, versionMinor: 0 };
+conversation.sendOptimistic(customContent, contentType);
+```
+
+```tsx [React Native]
+Code sample coming soon.
+```
+
+```kotlin [Kotlin]
+// Send a message optimistically (displays immediately in the sender's UI)
+conversation.sendOptimistic("Hello world")
+
+// For custom content types, specify the content type
+val customContent = mapOf("foo" to "bar")
+val contentType = ContentTypeId(
+    authorityId = "example",
+    typeId = "test",
+    versionMajor = 1,
+    versionMinor = 0
+)
+conversation.sendOptimistic(customContent, contentType)
+```
+
+```swift [Swift]
+// Send a message optimistically (displays immediately in the sender's UI)
+try await conversation.sendOptimistic("Hello world")
+
+// For custom content types, specify the content type
+let customContent = ["foo": "bar"]
+let contentType = ContentTypeId(
+    authorityId: "example",
+    typeId: "test",
+    versionMajor: 1,
+    versionMinor: 0
+)
+try await conversation.sendOptimistic(customContent, contentType: contentType)
+```
+
+:::
+
+### 2. Publish an optimistically sent message to the network
+
+After optimistically sending a message locally, use `publishMessages` to publish the message to the XMTP network so it can be delivered to recipients.
+
+Be sure to publish messages to the network in the order the user sent them. For example, if a message is still being published when a user attempts to optimistically send another message, wait for the current message to be published to the network before publishing the next message.
```

Comment:
i think this is all handled internally and order shouldn't matter since each message has a timestamp.